src/common/queries/account.ts

```diff
@@ -35,16 +35,53 @@ export const getInstalledModules = async ({
   })
 
   const responseBody = await response.json()
-
-  if (response.ok) {
-    return responseBody.data.SmartAccount_ModuleQuery.map(
-      (module: any) => module.module,
-    )
+  const modules = responseBody.data.SmartAccount_ModuleQuery.map(
+    (module: any) => module.moduleAddress,
+  )
+  // @Review: response.ok is still true even if:
+  // 1. Indexer does not support passed network
+  // 2. Smart account is not deployed or compatible
+  // This means that modules will be an empty array if something goes wrong
+  // @Recommendation: response.ok should be false if something goes wrong
+  if (response.ok && modules.length > 0) {
+    return modules
   } else {
-    throw new Error(
-      `Error: ${responseBody.errors
-        .map((error: any) => error.message)
-        .join(', ')}`,
-    )
+    const installedModules = await client.readContract({
+      address: account.address,
+      functionName: 'getValidatorsPaginated',
```

Comment:
fyi this will only get validators; one downside with this approach that its not possible to get all installed modules; eg nexus and safe only allow you to get validators, executors and (in the case of safe some) hooks, but not fallback modules